wagtail/contrib/settings/models.py

```diff
@@ -13,16 +13,63 @@ class BaseSetting(models.Model):
     :func:`~wagtail.contrib.settings.registry.register_setting`
     """
 
+    # Override to fetch ForeignKey values in the same query when
+    # retrieving settings via for_site()
+    select_related = None
+
     site = models.OneToOneField(
         Site, unique=True, db_index=True, editable=False, on_delete=models.CASCADE)
 
     class Meta:
         abstract = True
 
+    @classmethod
+    def base_queryset(cls):
+        """
+        Returns a queryset of objects of this type to use as a base
+        for calling get_or_create() on.
+
+        You can use the `select_related` attribute on your class to
+        specify a list of foreign key field names, which the method
+        will attempt to select additional related-object data for
+        when the query is executed.
+
+        If your needs are more complex than this, you can override
+        this method on your custom class.
+        """
+        queryset = cls.objects.all()
+        if cls.select_related is not None:
+            queryset = queryset.select_related(*cls.select_related)
+        return queryset
+
     @classmethod
     def for_site(cls, site):
         """
-        Get an instance of this setting for the site.
+        Get or create an instance of this setting for the site.
+        """
+        queryset = cls.base_queryset()
+        return queryset.get_or_create(site=site)[0]
```

Comment:
noticed this moves away from the `instance, created =  Model.get_or_create()` pattern. Would be good to be consistent. `get_or_create()[0]` is more cryptic